src/elements/common/annotator-context/withAnnotations.tsx

```diff
@@ -28,15 +31,41 @@ const defaultState: AnnotatorState = {
     meta: null,
 };
 
+type WithHistoryProps = {
+    history?: History;
+};
+
+type MatchParams = {
+    fileVersionId?: string;
+    annotationId?: string;
+};
+
+type ResultProps<P> = P & WithAnnotationsProps & WithHistoryProps;
+
 export default function withAnnotations<P extends object>(
     WrappedComponent: React.ComponentType<P>,
 ): WithAnnotationsComponent<P> {
-    class ComponentWithAnnotations extends React.Component<P & WithAnnotationsProps, AnnotatorState> {
+    class ComponentWithAnnotations extends React.Component<ResultProps<P>, AnnotatorState> {
         static displayName: string;
 
         annotator: Annotator | null = null;
 
-        state = defaultState;
+        constructor(props: ResultProps<P>) {
+            super(props);
+
+            // Determine by url if there is already a deeply linked annotation
+            const { history } = props;
+            const pathname = getProp(history, 'location.pathname');
```

Comment:
Can these be undefined/null? Does `matchPath` handle those values adequately?

Reply:
yeah, if there's no `history` (i.e. outside of the webapp) this could be undefined but `matchPath` handles it gracefully and returns `null`